lib/protocol/bep_extensions.go

```diff
@@ -213,6 +218,10 @@ func (f FileInfo) isEquivalent(other FileInfo, modTimeWindow time.Duration, igno
 		return false
 	}
 
+	if f.Hidden != other.Hidden {
```

Comment:
I haven't thought it through, but this might be problematic: A file that's hidden on windows but doesn't start with a dot will never be hidden on linux. That might just undo any hidden-ness on windows or even result in some nasty pull-scan loop.

Reply:
I think we need to take a step back and look at what exactly should be achieved here:

As is a file named 'foo' hidden on windows, that is pulled by a linux machine will be created, and then once scanning will see it's not hidden (no dot), and thus revert the hidden status. We can circumvent that by adding a check for windows in this line to prevent scanning on other systems to revert the hidden status. However that still leaves the problem that if the linux machine changes the file, it will then become unhidden again. That could be mitigated by adding yet another field to ignore the hidden status. Or make the hidden field an uint instead of bool, allowing for a third "undecided" state. However that feels like too much complexity for the gain and a "best effort" solution might make more sense: Create files with dot as hidden on windows, pick up the hidden attribute and send it as an optional parameter, that gets ignored (and dropped) by anyone but windowses and on windowses, preserve the hidden status when pulling a file without a hidden attribute.